src/main/java/seedu/notus/command/EditEventCommand.java

```diff
@@ -41,225 +47,356 @@
+    }
 
-        // Edit a reminder is indicated
-        if (!reminderTodo.isBlank()) {
-            HashMap<String, ArrayList<Integer>> originalReminderSchedule = event.getReminderPeriods();
-            switch (reminderTodo) {
-            case REMINDER_TYPE_ADD:
-                if (!event.getIsToRemind()) {
-                    event.setIsToRemind(true);
-                    event.setReminderPeriods(reminderSchedule);
-                    break;
-                }
-                warningSignalReminders = true;
-                for (String unit : reminderSchedule.keySet()) {
-                    ArrayList<Integer> remindersToAdd = reminderSchedule.get(unit);
-                    ArrayList<Integer> remindersPresent = originalReminderSchedule.get(unit);
-                    if (remindersPresent == null) {
-                        originalReminderSchedule.put(unit, remindersToAdd);
-                        warningSignalReminders = false;
-                        continue;
-                    }
-                    for (Integer reminder : remindersToAdd) {
-                        if (!remindersPresent.contains(reminder)) {
-                            remindersPresent.add(reminder);
-                            warningSignalReminders = false;
-                        }
-                    }
+    private String getStartEndTimeErrors(LocalDateTime startDateTime, LocalDateTime endDateTime) {
+        if (endDateTime.compareTo(startDateTime) < 0) {
+            return EDIT_EVENT_END_TIME_AFTER_START_WARNING;
+        }
+        if (endDateTime.getDayOfYear() != startDateTime.getDayOfYear()) {
+            return EDIT_EVENT_END_DATE_AFTER_START_DATE_WARNING;
+        }
+        return "";
+    }
+
+    /**
+     * Private method to edit both startDateTime and endDateTime of events. Will exit if none is edited. Will print
+     * errors if endDate is set to earlier than startDate, or endDate is on a different date from startDate.
+     *
+     * @param event Event to set on.
+     * @param results Results List to add on the result of this operation.
+     */
+    private void editTimings(Event event, ArrayList<String> results) {
+        // Edit startDate is indicated
+        if (newStartDateTime != null) {
+            if (newEndDateTime == null) {
+                LocalDateTime endDateTime = newStartDateTime.plusMinutes(event.getEventLengthInMinutes());
+                event.setStartDateTime(newStartDateTime);
+                if (endDateTime.getDayOfYear() != newStartDateTime.getDayOfYear()) {
+                    event.setEndDateTime(newStartDateTime.with(DEFAULT_EVENT_END_TIMING));
+                    results.add(EDIT_EVENT_START_TIME_SUCCESS_WARNING);
+                    return;
                 }
+                event.setEndDateTime(endDateTime);
+                results.add(EDIT_EVENT_START_TIME_SUCCESS_MESSAGE);
+                results.add(EDIT_EVENT_END_TIME_SHIFT_SUCCESS_MESSAGE);
+                return;
```

Comment:
Feels like too many nested if block. Can you consider an alternative way to get the same output?

Reply:
I tried removing it but I can't get the logic to work without the 3 nested ifs

Reply:
@R-Ramana can you check it now?